### ISSUE TITLE
ruff server does not support E30* errors?

### ISSUE THREAD

**Issue by a1401358759** [astral-sh/ruff#10814]
ruff version: 0.3.5
lsp ocnfig:
```lua
return {
  filetypes = { "python" },
  single_file_support = true,
  cmd = { "ruff", "server", "--preview", "--config", "~/.config/nvim/pyproject.toml" },
  settings = {
    args = options.ruff_args,
  },
}
```
pyproject.tmol
```toml
[tool.ruff.lint]
# Enable Pyflakes (`F`) and a subset of the pycodestyle (`E`)  codes by default.
# Unlike Flake8, Ruff doesn't enable pycodestyle warnings (`W`) or
# McCabe complexity (`C901`) by default.
select = ["E4", "E7", "E9", "F"]
ignore = ["E402","E501","N802","N803","N806","N801","N813","N815","N816","RUF001","RUF002","RUF003","RUF012","ARG001"]
extend-select = ["E", "N", "W", "ARG", "RUF"]

# Allow fix for all enabled rules (when `--fix`) is provided.
fixable = ["ALL"]
unfixable = []

# Allow unused variables when underscore-prefixed.
dummy-variable-rgx = "^(_+|(_+[a-zA-Z0-9_]*[a-zA-Z0-9]+?))$"
```
but i did not get any errors
<img width="384" alt="image" src="https://github.com/astral-sh/ruff/assets/15830146/0fcae23e-98d3-4576-b1ee-c3177a28934b">

**Comment by charliermarsh** [comment]
Can you try adding `preview = true` to your `pyproject.toml`, under `[tool.ruff]`?

**Comment by a1401358759** [comment]
@charliermarsh 
```toml
[tool.ruff]
preview = true
# Exclude a variety of commonly ignored directories.
exclude = [
    ".bzr",
    ".direnv",
    ".eggs",
    ".git",
    ".git-rewrite",
    ".hg",
    ".ipynb_checkpoints",
    ".mypy_cache",
    ".nox",
    ".pants.d",
    ".pyenv",
    ".pytest_cache",
    ".pytype",
    ".ruff_cache",
    ".svn",
    ".tox",
    ".venv",
    ".vscode",
    "__pypackages__",
    "_build",
    "buck-out",
    "build",
    "dist",
    "node_modules",
    "site-packages",
    "venv",
]

# Same as Black.
line-length = 120
indent-width = 4
```
It's still not working.

**Comment by MichaReiser** [comment]
@a1401358759 did you keep your existing lint configuration?

```toml
[tool.ruff]
preview = true
# Exclude a variety of commonly ignored directories.
exclude = [
    ".bzr",
    ".direnv",
    ".eggs",
    ".git",
    ".git-rewrite",
    ".hg",
    ".ipynb_checkpoints",
    ".mypy_cache",
    ".nox",
    ".pants.d",
    ".pyenv",
    ".pytest_cache",
    ".pytype",
    ".ruff_cache",
    ".svn",
    ".tox",
    ".venv",
    ".vscode",
    "__pypackages__",
    "_build",
    "buck-out",
    "build",
    "dist",
    "node_modules",
    "site-packages",
    "venv",
]

# Same as Black.
line-length = 120
indent-width = 4

[tool.ruff.lint]

# Enable Pyflakes (`F`) and a subset of the pycodestyle (`E`)  codes by default.
# Unlike Flake8, Ruff doesn't enable pycodestyle warnings (`W`) or
# McCabe complexity (`C901`) by default.
select = ["E4", "E7", "E9", "F"]
ignore = ["E402","E501","N802","N803","N806","N801","N813","N815","N816","RUF001","RUF002","RUF003","RUF012","ARG001"]
extend-select = ["E", "N", "W", "ARG", "RUF"]

# Allow fix for all enabled rules (when `--fix`) is provided.
fixable = ["ALL"]
unfixable = []

# Allow unused variables when underscore-prefixed.
dummy-variable-rgx = "^(_+|(_+[a-zA-Z0-9_]*[a-zA-Z0-9]+?))$"
```

**Comment by a1401358759** [comment]
@MichaReiser this is my whole config
```lua
[tool.ruff]
preview = true
# Exclude a variety of commonly ignored directories.
exclude = [
    ".bzr",
    ".direnv",
    ".eggs",
    ".git",
    ".git-rewrite",
    ".hg",
    ".ipynb_checkpoints",
    ".mypy_cache",
    ".nox",
    ".pants.d",
    ".pyenv",
    ".pytest_cache",
    ".pytype",
    ".ruff_cache",
    ".svn",
    ".tox",
    ".venv",
    ".vscode",
    "__pypackages__",
    "_build",
    "buck-out",
    "build",
    "dist",
    "node_modules",
    "site-packages",
    "venv",
]

# Same as Black.
line-length = 120
indent-width = 4

[tool.ruff.lint]
# Enable Pyflakes (`F`) and a subset of the pycodestyle (`E`)  codes by default.
# Unlike Flake8, Ruff doesn't enable pycodestyle warnings (`W`) or
# McCabe complexity (`C901`) by default.
select = ["E4", "E7", "E9", "F"]
ignore = ["E402","E501","N802","N803","N806","N801","N813","N815","N816","RUF001","RUF002","RUF003","RUF012","ARG001"]
extend-select = ["E", "N", "W", "ARG", "RUF"]

# Allow fix for all enabled rules (when `--fix`) is provided.
fixable = ["ALL"]
unfixable = []

# Allow unused variables when underscore-prefixed.
dummy-variable-rgx = "^(_+|(_+[a-zA-Z0-9_]*[a-zA-Z0-9]+?))$"

[tool.ruff.format]
# Like Black, use double quotes for strings.
quote-style = "double"

# Like Black, indent with spaces, rather than tabs.
indent-style = "space"

# Like Black, respect magic trailing commas.
skip-magic-trailing-comma = false

# Like Black, automatically detect the appropriate line ending.
line-ending = "auto"

# Enable auto-formatting of code examples in docstrings. Markdown,
# reStructuredText code/literal blocks and doctests are all supported.
#
# This is currently disabled by default, but it is planned for this
# to be opt-out in the future.
docstring-code-format = false

# Set the line length limit used when formatting code snippets in
# docstrings.
#
# This only has an effect when the `docstring-code-format` setting is
# enabled.
docstring-code-line-length = "dynamic"
```

**Comment by MichaReiser** [comment]
Hm okay. This works for me but I use the old LSP (and vscode). 

Can you try removing the 
```
settings = {
    args = options.ruff_args,
  },
```

because the new server doesn't yet support options (but I'm just guessing at this point).

@snowsignal would you mind helping out?

**Comment by a1401358759** [comment]
@MichaReiser  I try removing the
```lua
settings = {
    args = options.ruff_args,
  },
```
but it's still not working,ruff-lsp works good for me.

**Comment by dhruvmanila** [comment]
Hey @a1401358759, I don't think the new server supporst configuring Ruff via the global `--config` flag or a local configuration file.

@snowsignal Correct me if I'm wrong here.

**Comment by snowsignal** [comment]
@a1401358759 Right now the new server has a few limitations, especially around configuration. At the moment, we don't support configuring the server with the `--config` flag, nor do we support the `args` field in `settings` (like @dhruvmanila and @MichaReiser mentioned). Alternative configuration is something we're actively working on, but at the moment the configuration just defaults to any `pyproject.toml` / `ruff.toml` files at the root of your workspace or in parent folders.

Thank you for opening this issue - I'll follow up once we have a setting like this available.

**Comment by a1401358759** [comment]
@snowsignal Okay, thank you very much for your answer.

**Comment by snowsignal** [comment]
@a1401358759 Once https://github.com/astral-sh/ruff/pull/11225 lands in the next release, you should be able to specify a custom configuration file!

Here's what your LSP config could look like:
```lua
return {
  filetypes = { "python" },
  single_file_support = true,
  cmd = { "ruff", "server", "--preview", },
  settings = {
    configuration = "~/.config/nvim/pyproject.toml"
  },
}
```

The `configuration` setting is what you should use to specify the configuration file.

Let me know if you run into any more issues!

**Comment by a1401358759** [comment]
my lsp config
```lua
return {
  filetypes = { "python" },
  single_file_support = true,
  cmd = { "ruff", "server", "--preview" },
  settings = {
    configuration = "~/.config/nvim/pyproject.toml",
  },
}
```

my pyprohect.toml
```toml
[tool.ruff]
preview = true
# Exclude a variety of commonly ignored directories.
exclude = [
    ".bzr",
    ".direnv",
    ".eggs",
    ".git",
    ".git-rewrite",
    ".hg",
    ".ipynb_checkpoints",
    ".mypy_cache",
    ".nox",
    ".pants.d",
    ".pyenv",
    ".pytest_cache",
    ".pytype",
    ".ruff_cache",
    ".svn",
    ".tox",
    ".venv",
    ".vscode",
    "__pypackages__",
    "_build",
    "buck-out",
    "build",
    "dist",
    "node_modules",
    "site-packages",
    "venv",
]

# Same as Black.
line-length = 120
indent-width = 4

[tool.ruff.lint]
# Enable Pyflakes (`F`) and a subset of the pycodestyle (`E`)  codes by default.
# Unlike Flake8, Ruff doesn't enable pycodestyle warnings (`W`) or
# McCabe complexity (`C901`) by default.
select = ["E4", "E7", "E9", "F"]
ignore = ["E402","E501","N802","N803","N806","N801","N813","N815","N816","RUF001","RUF002","RUF003","RUF012","ARG001"]
extend-select = ["E", "N", "W", "ARG", "RUF"]

# Allow fix for all enabled rules (when `--fix`) is provided.
fixable = ["ALL"]
unfixable = []

# Allow unused variables when underscore-prefixed.
dummy-variable-rgx = "^(_+|(_+[a-zA-Z0-9_]*[a-zA-Z0-9]+?))$"

[tool.ruff.format]
# Like Black, use double quotes for strings.
quote-style = "double"

# Like Black, indent with spaces, rather than tabs.
indent-style = "space"

# Like Black, respect magic trailing commas.
skip-magic-trailing-comma = false

# Like Black, automatically detect the appropriate line ending.
line-ending = "auto"

# Enable auto-formatting of code examples in docstrings. Markdown,
# reStructuredText code/literal blocks and doctests are all supported.
#
# This is currently disabled by default, but it is planned for this
# to be opt-out in the future.
docstring-code-format = false

# Set the line length limit used when formatting code snippets in
# docstrings.
#
# This only has an effect when the `docstring-code-format` setting is
# enabled.
docstring-code-line-length = "dynamic"
```

but ruff server still doesn't check E30* errors

<img width="611" alt="image" src="https://github.com/astral-sh/ruff/assets/15830146/653cd1af-6218-434d-97c6-88ef74e0f674">

Is there still a problem with my configuration?
